core/src/main/java/org/apache/brooklyn/core/catalog/internal/CatalogBundleLoader.java

```diff
@@ -103,8 +103,8 @@ private TemporaryInternalScanResult scanForCatalogInternal(Bundle bundle, boolea
                 this.managementContext.getCatalog().addTypesFromBundleBom(bomText, mb, force, result.mapOfNewToReplaced);
                 if (validate) {
                     Set<RegisteredType> matches = MutableSet.copyOf(this.managementContext.getTypeRegistry().getMatching(RegisteredTypePredicates.containingBundle(mb.getVersionedName())));
-                    if (!(matches.containsAll(result.mapOfNewToReplaced.keySet()) && result.mapOfNewToReplaced.keySet().containsAll(matches))) {
-                        // sanity check
+                    if (matches.size()!=result.mapOfNewToReplaced.size()) {
```

Comment:
I'm afraid this will be too loose: the 2 sets have the same size but we don't know if the items are "the same", i.e. same `symbolicName` and `version`.

Could we have a middle ground here? Something like looping over each items in `result.mapOfNewToReplaced`, checking if `matches` contains one with the same `symbolicName` and `version`. Those fields should be resolved and would provide sufficient equality in this case. WDYT?

Reply:
@ahgittin can you explain more the purpose/context of this check, and why you only want to log.warn if the sizes are different?

I recall seeing such warnings in the log when I suspect we shouldn't have, so think you're right to guard against it better.

For @tbouron's suggestion of checking the symbolicName:version, that makes sense - or does this get called a lot or with really big things so performance matters?